### ISSUE TITLE
How to use ethernet_mac repository ?

### ISSUE THREAD

**Issue by liubenyuan** [pabennett/ethernet_mac#1]
Hi,

I am using your uart module in my project (transmitting data via RN-42 or CP2102 on my lx9 microboard, i place them in (https://github.com/liubenyuan/lx9health) ), it is great !!

Now I want to transmite data via the ethernet protocol, still based on the lx9 microboard with NS DP83848J PHY, could you please tell me how to use your module in the design?

I am glad to test it on Spartan 6 series.

Best!

liu benyuan

**Comment by pabennett** [comment]
Hi Liu,

I'm glad the UART is working for you. The ethernet MAC in this repository is not a complete solution; I developed it as far as receiving and verifying frames before moving on to other things and I left it here as a reference to others.

My colleague dawsonjon has a working implementation of a gigabit ethernet MAC here: https://github.com/dawsonjon/Chips-Demo, although it wont be compatible with your LX9 microboard, which only supports 10/100. If you wanted a challenge you could use the PHY I/F from my repository and try tying it together with Jon's work to make a solution for your board.

Pete

**Comment by liubenyuan** [comment]
Hi, Pete,

So the downstream is working ? May I test the **receiving and verifying** using the code ? 

Best!

Liu.

**Comment by pabennett** [comment]
Yes, the ethernet_receive component can do that for you. It connects to the PHY, validates the CRC of received frames and then passes the frame payload to an upper layer (internet_receive).

**Comment by liubenyuan** [comment]
BTW, I read your code and found that your are using MII mode of PHY device, right?

**Comment by pabennett** [comment]
Yes that's right, just RX though.
